packages/ui/molecules/AnchorNav/AnchorNavTarget.ts

```diff
@@ -0,0 +1,32 @@
+import { Base, withIntersectionObserver } from '@studiometa/js-toolkit';
+import type { BaseProps, BaseConfig } from '@studiometa/js-toolkit';
+
+export interface AnchorNavTargetProps extends BaseProps {
+  $refs: {
+    item: HTMLElement[];
+  };
+}
+
+/**
+ * Manage a sticky table section.
+ */
+export class AnchorNavTarget extends withIntersectionObserver(Base, {
```

Comment:
**question:** Any specific reason to use `withIntersectionObserver` rather than `withMountWhenInView`? 

The advantage of the `withMountWhenInView` decorator is that you do not have to add a custom event, you can listen to the `mounted` event. This would change the `onAnchorNavTargetIsIntersected` method from `AnchorNav` to `onAnchorNavTargetMounted` with the following implementation:

```js
onAnchorNavTargetMounted(index) {
  const { id } = this.$children.AnchorNavTarget[index];

  for (const anchorNavLink of this.$children.AnchorNavLink) {
    const method = id === anchorNavLink.targetId ? 'enter' : 'leave';
    anchorNavLink[method]();
  }
}
```

> [!NOTE]
> The `targetId` getter would need to be added to the `AnchorNavLink` class

We could even go further and only trigger the `leave` method when the `AnchorNavTarget` is destroyed (when out of view). This would allow for multiple `AnchorNavLink` instances being active simultaneously. For this behavior, the following implementation in `AnchorNav` should work: 

```js
onAnchorNavTargetMounted(index) {
  const { id } = this.$children.AnchorNavTarget[index];

  for (const anchorNavLink of this.$children.AnchorNavLink) {
    if (id === anchorNavLink.targetId) {
      anchorNavLink.enter();
      break;
    }
  }
}

onAnchorNavTargetDestroyed(index) {
  const { id } = this.$children.AnchorNavTarget[index];

  for (const anchorNavLink of this.$children.AnchorNavLink) {
    if (id === anchorNavLink.targetId) {
      anchorNavLink.leave();
      break;
    }
  }
}
```